src/Interpreters/executeSelectQuery.cpp

```diff
@@ -59,8 +59,7 @@ void executeSelectQuery(const String & query, ContextPtr query_context, const st
     }
     else
     {
-        /// Pass a query which is not select query
-        assert(false);
+        io.onFinish();
```

Comment:
I think we shall move `io.onFinish()` to the end of this function which means in all cases, we will need call `io.onFinish()` ?

Reply:
I think only when non-selelct or show sql statements are needed io.onFinish() , but  move io.onFinish() to the end of this function will be more safe